downloads/swan-lake-release-notes/swan-lake-preview8/RELEASE_NOTE.md

```diff
@@ -0,0 +1,519 @@
+
+    ```ballerina
+    public function main() {
+        studentTotalMarks(90, 85, 75);
+        studentTotalMarks(85, 85, 75, english = 75);
+        studentTotalMarks(75, 85, 90, english = 80, generalTest = 70);
+    }
+    ```
+
+##### Service Typing Changes
+
+Now, services are being based on objects. The service declaration syntax below is mere syntactic sugar for creating a new instance of a service class and then attaching it to a listener. With this change, the path that the service should serve on can be provided in the service-declaration syntax. This used to be provided using an annotation before.
+
+Therefore, the example below;
+
+```ballerina
+import ballerina/http;
+service hello on new http:Listener(9090) {
+
+    resource function sayHello() returns error? {
+
+    }
+}
+```
+
+will turn into the code below.
+
+```ballerina
+import ballerina/http;
+service [/optional_base_path] on new http:Listener(9090) {
+
+    resource function get sayHello() returns string {
+
+    }
+}
+```
+
+>**Note:** There is no longer a service variable name as `hello` in the first example. 
+Since services are objects, they can contain fields and regular methods. In addition to that, services can contain remote methods and resource methods.
+
+Resource functions are defined as follows:
+
+ `resource function` `resource-method-name` `resource-path``() { }`
+
+- Resource functions do not have a method name similar to a regular method.
+- The `resource-method-name` instructs the listener on which operations this resource allows. For example, in the HTTP listener, the `resource-method-name` maps to the HTTP methods such as `GET` or `PUT`.
+- The `resource-path` is the path in which this resource resides within the service. You can use `.` as the `resource-path` to indicate the path `/`.
+
+A service declared using the service-declaration syntax cannot be referred to using an identifier anymore. Instead, you need to use an object constructor prepended by the `service` keyword to declare a service object or create a new instance of a service class.
+
+```ballerina
+service object { } hello = service object {
+    string message = "Hello";
+    resource function get sayHello() returns string {
+
+    }
+
+}
+listener http:Listener l = new(7000);
+string[] basePath = ["/"]; // service base path: /
+l.attach(hello, basePath);
+```
+
+```ballerina
+service class HelloService {
+    string message;
+    resource function get sayHello() returns string {
+
+    }
+
+    function init(string message) {
+        self.message = message;
+    }
+}
+var helo = new HelloService();
```

Comment:
Doesn't the constructor need string arg?

Reply:
Yes thank you for pointing